modules/apps/portal-odata/portal-odata-impl/src/main/java/com/liferay/portal/odata/internal/sort/SortParserImpl.java

```diff
@@ -96,6 +96,10 @@ public List<SortField> parse(String sortString) {
 	protected Optional<EntityField> getEntityFieldOptional(
 		Map<String, EntityField> entityFieldsMap, String fieldName) {
 
+		if (Objects.equals(fieldName, "externalReferenceCode")) {
```

Comment:
should find out the root cause to avoid hard code, and still having bug when clicking on the other column to filter